torch/nn/modules/module.py

```diff
@@ -965,18 +965,20 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
             key = prefix + name
             if key in state_dict:
                 input_param = state_dict[key]
-
+                # This is used to avoid copying uninitialized parameters into
+                # non-lazy modules, since they dont have the hook to do the checks
+                # in such case, it will error when accessing the .shape attribute.
+                is_param_lazy = isinstance(param, torch.nn.parameter.UninitializedParameter)
```

Comment:
So you rely on the `.copy_()` below failing and providing the error message here?

Reply:
When accessing the lazy attribute .shape it will hard-error, the rest of errors are already covered by the hook.
If I can't access a parameter data, it becomes quite tricky to cover all the cases with simple code.

Reply:
Like @albanD I'm a little confused by this code, too. Is there a simpler or more direct way to write this check, possibly by testing if the param is lazy vs. testing if it's not lazy twice?

Reply:
Alternatively, if lazy modules don't support state dicts maybe this code can just be removed.

Reply:
Following our other discussions, I think this is OK for now, but we should revisit this approach in the near future.